cmake/CompileFlags.cmake

```diff
@@ -72,22 +70,6 @@ macro(compile_flags)
             -fno-strict-aliasing
         )
 
-        # When we are a stable release (Release build + USE_ASSERTS not set),
-        # assertations are off, which trigger a lot of warnings. We disable
-        # these warnings for these releases.
-        if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-            add_compile_options(
-                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
-                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-parameter>"
-                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-variable>"
-            )
-        else()
-            add_compile_options(
-                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
-                "$<${IS_STABLE_RELEASE}:-Wno-unused-parameter>"
-            )
-        endif()
-
```

Comment:
We don't actually have a mechanic to test releases without asserts before the actual release, and I guess this is why they exist. These warnings would happen so late in our release process, that it is a lot of effort to correct for them.

This change kinda implies we should add to the CI a run that has asserts disabled, I guess. Without that, this is a tricky change (as warnings can go unnoticed for a long time).

Reply:
https://github.com/OpenTTD/OpenTTD/pull/11309

Would address that concern.